auslib/admin/views/history.py

```diff
@@ -54,43 +44,54 @@ def get(self, type_, change_id, field):
         return Response(value, content_type='text/plain')
 
 
-class DiffView(FieldView):
-    """/diff/:type/:id/:field"""
-
-    def get_prev_id(self, value, change_id):
-
-        release_name = value['name']
-
-        table = dbo.releases.history
-        old_revision = table.select(
-            where=[
-                table.name == release_name,
-                table.change_id < change_id,
-                table.data_version != null()
-            ],
-            limit=1,
-            order_by=[table.timestamp.desc()],
-        )
-
-        return old_revision[0]['change_id']
-
-    def get(self, type_, change_id, field):
-        value = self.get_value(type_, change_id, field)
-        data_version = self.get_value(type_, change_id, "data_version")
-
-        prev_id = self.get_prev_id(value, change_id)
-        previous = self.get_value(type_, prev_id, field)
-        prev_data_version = self.get_value(type_, prev_id, "data_version")
-
-        value = self.format_value(value)
-        previous = self.format_value(previous)
-
-        result = difflib.unified_diff(
-            previous.splitlines(),
-            value.splitlines(),
-            fromfile="Data Version {}".format(prev_data_version),
-            tofile="Data Version {}".format(data_version),
-            lineterm=""
-        )
-
-        return Response('\n'.join(result), content_type='text/plain')
+class HistoryViewParameters(object):
+    def __init__(self,
+                 get_object_callback,
```

Comment:
That's a great point - it's much explicit with the callbacks. Let's stick with that!